File: suc/kotlin/034.kt
package com.example.singlefileui

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import androidx.core.view.WindowCompat
import androidx.core.view.WindowInsetsCompat
import androidx.core.view.WindowInsetsControllerCompat

private const val NAME = "010*ExploreMap*en"
private const val UI_TYPE = "Social"
private const val STYLE_THEME = "Monochrome Terminal"
private const val LANG = "en"
private const val BASELINE_SIZE = "720x1280"

object AppTokens {
    object Colors {
        val primary = Color(0xFF00FF66)
        val secondary = Color(0xFFCCCCCC)
        val tertiary = Color(0xFF999999)
        val background = Color(0xFF000000)
        val surface = Color(0xFF0A0A0A)
        val surfaceVariant = Color(0xFF111111)
        val outline = Color(0xFF1F1F1F)
        val success = Color(0xFF16A34A)
        val warning = Color(0xFFF59E0B)
        val error = Color(0xFFFF4444)
        val onPrimary = Color(0xFF000000)
        val onSecondary = Color(0xFFFFFFFF)
        val onTertiary = Color(0xFFFFFFFF)
        val onBackground = Color(0xFFE0E0E0)
        val onSurface = Color(0xFFE0E0E0)
    }
    object TypographyTokens {
        val display = TextStyle(fontSize = 28.sp, fontWeight = FontWeight.Bold)
        val headline = TextStyle(fontSize = 20.sp, fontWeight = FontWeight.Medium)
        val title = TextStyle(fontSize = 16.sp, fontWeight = FontWeight.Medium)
        val body = TextStyle(fontSize = 13.sp, fontWeight = FontWeight.Normal)
        val label = TextStyle(fontSize = 11.sp, fontWeight = FontWeight.Medium)
    }
    object Shapes {
        val small = RoundedCornerShape(6.dp)
        val medium = RoundedCornerShape(10.dp)
        val large = RoundedCornerShape(14.dp)
    }
    object Spacing {
        val xs = 4.dp
        val sm = 8.dp
        val md = 12.dp
        val lg = 16.dp
        val xl = 24.dp
        val xxl = 32.dp
    }
    data class ShadowSpec(val elevation: Dp, val radius: Dp, val dy: Dp, val opacity: Float)
    object ElevationMapping {
        val level1 = ShadowSpec(0.dp, 0.dp, 0.dp, 0f)
        val level2 = ShadowSpec(2.dp, 4.dp, 2.dp, 0.12f)
    }
}

private val AppColorScheme = darkColorScheme(
    primary = AppTokens.Colors.primary,
    onPrimary = AppTokens.Colors.onPrimary,
    secondary = AppTokens.Colors.secondary,
    onSecondary = AppTokens.Colors.onSecondary,
    background = AppTokens.Colors.background,
    onBackground = AppTokens.Colors.onBackground,
    surface = AppTokens.Colors.surface,
    onSurface = AppTokens.Colors.onSurface,
    surfaceVariant = AppTokens.Colors.surfaceVariant,
    outline = AppTokens.Colors.outline,
    error = AppTokens.Colors.error
)

private val AppTypography = Typography(
    displayLarge = AppTokens.TypographyTokens.display,
    headlineMedium = AppTokens.TypographyTokens.headline,
    titleMedium = AppTokens.TypographyTokens.title,
    bodyMedium = AppTokens.TypographyTokens.body,
    labelMedium = AppTokens.TypographyTokens.label
)

@Composable
fun AppTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = AppColorScheme,
        typography = AppTypography,
        shapes = Shapes(
            small = AppTokens.Shapes.small,
            medium = AppTokens.Shapes.medium,
            large = AppTokens.Shapes.large
        ),
        content = content
    )
}

data class MapMarker(val name: String, val distance: String)

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun RootScreen() {
    var query by remember { mutableStateOf("") }
    val markers = remember {
        listOf(
            MapMarker("Central Plaza", "1.2 km"),
            MapMarker("Museum District", "3.4 km"),
            MapMarker("Tech Park", "5.6 km"),
            MapMarker("Old Town Gate", "7.8 km")
        )
    }
    Scaffold(
        contentWindowInsets = WindowInsets(0),
        topBar = {
            CenterAlignedTopAppBar(
                title = {
                    Text(
                        "Explore Map",
                        style = MaterialTheme.typography.displayLarge,
                        color = AppTokens.Colors.primary
                    )
                },
                colors = TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = AppTokens.Colors.background)
            )
        },
        containerColor = AppTokens.Colors.background
    ) { pad ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(pad)
                .background(AppTokens.Colors.background)
                .padding(AppTokens.Spacing.lg),
            verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.lg)
        ) {
            OutlinedTextField(
                value = query,
                onValueChange = { query = it },
                label = { Text("Search destination", color = AppTokens.Colors.onSurface) },
                singleLine = true,
                shape = AppTokens.Shapes.medium,
                colors = OutlinedTextFieldDefaults.colors(
                    focusedTextColor = AppTokens.Colors.onSurface,
                    unfocusedTextColor = AppTokens.Colors.onSurface,
                    cursorColor = AppTokens.Colors.primary,
                    focusedBorderColor = AppTokens.Colors.primary,
                    unfocusedBorderColor = AppTokens.Colors.outline
                ),
                modifier = Modifier.fillMaxWidth()
            )
            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(AppTokens.Spacing.md)
            ) {
                markers.forEach {
                    Card(
                        modifier = Modifier.fillMaxWidth(),
                        shape = AppTokens.Shapes.medium,
                        colors = CardDefaults.cardColors(containerColor = AppTokens.Colors.surface),
                        elevation = CardDefaults.cardElevation(defaultElevation = AppTokens.ElevationMapping.level2.elevation)
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(AppTokens.Spacing.md),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Column {
                                Text(it.name, style = MaterialTheme.typography.titleMedium, color = AppTokens.Colors.onSurface)
                                Text(it.distance, style = MaterialTheme.typography.bodyMedium, color = AppTokens.Colors.secondary)
                            }
                            Button(
                                onClick = {},
                                shape = AppTokens.Shapes.small,
                                colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary, contentColor = AppTokens.Colors.onPrimary)
                            ) {
                                Text("View", style = MaterialTheme.typography.labelMedium)
                            }
                        }
                    }
                }
            }
            Spacer(modifier = Modifier.height(AppTokens.Spacing.xxl))
            Button(
                onClick = {},
                modifier = Modifier.fillMaxWidth().height(48.dp),
                shape = AppTokens.Shapes.large,
                colors = ButtonDefaults.buttonColors(containerColor = AppTokens.Colors.primary, contentColor = AppTokens.Colors.onPrimary)
            ) {
                Text("Refresh Map", style = MaterialTheme.typography.titleMedium)
            }
        }
    }
}

class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowCompat.setDecorFitsSystemWindows(window, false)
        val controller = WindowInsetsControllerCompat(window, window.decorView)
        controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
        controller.hide(WindowInsetsCompat.Type.systemBars())
        setContent {
            AppTheme {
                Surface(color = MaterialTheme.colorScheme.background) {
                    RootScreen()
                }
            }
        }
    }
    override fun onWindowFocusChanged(hasFocus: Boolean) {
        super.onWindowFocusChanged(hasFocus)
        if (hasFocus) {
            val controller = WindowInsetsControllerCompat(window, window.decorView)
            controller.systemBarsBehavior = WindowInsetsControllerCompat.BEHAVIOR_SHOW_TRANSIENT_BARS_BY_SWIPE
            controller.hide(WindowInsetsCompat.Type.systemBars())
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFF000000)
@Composable
fun PreviewScreen() {
    AppTheme {
        RootScreen()
    }
}

File: suc/swift/034.swift

import SwiftUI

// MARK: - AppTokens
// 严格对应 Kotlin 中的 AppTokens 定义，使用 CGFloat 替代 Dp，使用 Swift 的 Color 和 Font
struct AppTokens {
    struct Colors {
        static let primary = Color(hex: 0x00FF66)
        static let secondary = Color(hex: 0xCCCCCC)
        static let tertiary = Color(hex: 0x999999)
        static let background = Color(hex: 0x000000)
        static let surface = Color(hex: 0x0A0A0A)
        static let surfaceVariant = Color(hex: 0x111111)
        static let outline = Color(hex: 0x1F1F1F)
        static let success = Color(hex: 0x16A34A)
        static let warning = Color(hex: 0xF59E0B)
        static let error = Color(hex: 0xFF4444)
        static let onPrimary = Color(hex: 0x000000)
        static let onSecondary = Color(hex: 0xFFFFFFFF)
        static let onTertiary = Color(hex: 0xFFFFFFFF)
        static let onBackground = Color(hex: 0xE0E0E0)
        static let onSurface = Color(hex: 0xE0E0E0)
    }

    struct TypographyTokens {
        static let display = Font.system(size: 28, weight: .bold)
        static let headline = Font.system(size: 20, weight: .medium)
        static let title = Font.system(size: 16, weight: .medium)
        static let body = Font.system(size: 13, weight: .regular) // Normal -> regular
        static let label = Font.system(size: 11, weight: .medium)
    }

    struct Shapes {
        static let small: CGFloat = 6
        static let medium: CGFloat = 10
        static let large: CGFloat = 14
    }

    struct Spacing {
        static let xs: CGFloat = 4
        static let sm: CGFloat = 8
        static let md: CGFloat = 12
        static let lg: CGFloat = 16
        static let xl: CGFloat = 24
        static let xxl: CGFloat = 32
    }

    struct ShadowSpec {
        // Kotlin elevation is 0.dp, radius 4.dp, dy 2.dp, opacity 0.12f
        // SwiftUI shadow takes radius (blur), x/y offset, and opacity.
        // Mapping Compose radius to SwiftUI blur radius, Compose dy to SwiftUI y offset.
        let radius: CGFloat // Blur radius
        let y: CGFloat      // Y offset
        let opacity: Double
    }

    struct ElevationMapping {
        static let level1 = ShadowSpec(radius: 0, y: 0, opacity: 0)
        // Kotlin: elevation = 2.dp, radius = 4.dp, dy = 2.dp, opacity = 0.12f
        static let level2 = ShadowSpec(radius: 4, y: 2, opacity: 0.12)
    }
}

// MARK: - Color Extension for Hex Initialization
// 方便地从十六进制值创建 Color
extension Color {
    init(hex: UInt, alpha: Double = 1.0) {
        self.init(
            .sRGB,
            red: Double((hex >> 16) & 0xFF) / 255.0,
            green: Double((hex >> 8) & 0xFF) / 255.0,
            blue: Double(hex & 0xFF) / 255.0,
            opacity: alpha
        )
    }
}

// MARK: - Data Model
// 对应 Kotlin 的 MapMarker data class
struct MapMarker: Identifiable {
    let id = UUID() // SwiftUI ForEach 需要 Identifiable
    let name: String
    let distance: String
}

// MARK: - RootScreen View
struct RootScreen: View {
    @State private var query: String = ""
    @FocusState private var isSearchFieldFocused: Bool // 用于 OutlinedTextField 的焦点状态

    // 对应 Kotlin 中的 markers 列表
    let markers = [
        MapMarker(name: "Central Plaza", distance: "1.2 km"),
        MapMarker(name: "Museum District", distance: "3.4 km"),
        MapMarker(name: "Tech Park", distance: "5.6 km"),
        MapMarker(name: "Old Town Gate", distance: "7.8 km")
    ]

    var body: some View {
        ZStack { // 使用 ZStack 作为根容器，以便设置背景色并忽略安全区
            AppTokens.Colors.background
                .ignoresSafeArea() // 确保背景色覆盖整个屏幕，包括安全区

            VStack(spacing: 0) { // 主 VStack 布局，spacing: 0 避免 VStack 自身添加间距
                // Top Bar (对应 Kotlin 的 CenterAlignedTopAppBar)
                Text("Explore Map")
                    .font(AppTokens.TypographyTokens.display)
                    .foregroundColor(AppTokens.Colors.primary)
                    .frame(maxWidth: .infinity) // 填充宽度
                    .padding(.vertical, AppTokens.Spacing.lg) // 垂直内边距，模拟 TopAppBar 默认高度
                    .background(AppTokens.Colors.background)

                // 主内容区域
                VStack(spacing: AppTokens.Spacing.lg) { // 对应 Kotlin Column 的 verticalArrangement.spacedBy(AppTokens.Spacing.lg)
                    // Search Field (对应 Kotlin 的 OutlinedTextField)
                    TextField("Search destination", text: $query)
                        .focused($isSearchFieldFocused) // 绑定焦点状态
                        .font(AppTokens.TypographyTokens.body) // 文本和占位符字体
                        .foregroundColor(AppTokens.Colors.onSurface) // 文本颜色
                        .accentColor(AppTokens.Colors.primary) // 光标颜色
                        .padding(AppTokens.Spacing.md) // 内部文本内边距
                        .background(AppTokens.Colors.surfaceVariant) // 对应 Kotlin OutlinedTextField 的背景色
                        .cornerRadius(AppTokens.Shapes.medium) // 圆角
                        .overlay( // 边框
                            RoundedRectangle(cornerRadius: AppTokens.Shapes.medium)
                                .stroke(isSearchFieldFocused ? AppTokens.Colors.primary : AppTokens.Colors.outline, lineWidth: 1)
                        )
                        .frame(maxWidth: .infinity) // 填充宽度

                    // Markers List (对应 Kotlin 的 Column 包含 Card 列表)
                    VStack(spacing: AppTokens.Spacing.md) { // 对应 Kotlin Column 的 verticalArrangement.spacedBy(AppTokens.Spacing.md)
                        ForEach(markers) { marker in
                            CardView(marker: marker)
                        }
                    }
                    .frame(maxWidth: .infinity) // 填充宽度

                    Spacer() // 将底部的按钮推到底部

                    // Refresh Map Button (对应 Kotlin 的 Button)
                    Button(action: {
                        // 处理刷新地图操作
                        print("Refresh Map Tapped")
                    }) {
                        Text("Refresh Map")
                            .font(AppTokens.TypographyTokens.title)
                            .foregroundColor(AppTokens.Colors.onPrimary)
                            .frame(maxWidth: .infinity) // 填充宽度
                            .frame(height: 48) // 固定高度
                            .background(AppTokens.Colors.primary)
                            .cornerRadius(AppTokens.Shapes.large) // 圆角
                    }
                }
                .padding(AppTokens.Spacing.lg) // 对应 Kotlin Column 的 .padding(AppTokens.Spacing.lg)
            }
        }
        .statusBarHidden(true) // 隐藏顶部状态栏，满足全屏要求
    }
}

// MARK: - CardView
// 对应 Kotlin 中的 Card 及其内容
struct CardView: View {
    let marker: MapMarker

    var body: some View {
        HStack(alignment: .center, spacing: 0) { // 对应 Kotlin Row 的 horizontalArrangement.SpaceBetween 和 verticalAlignment.CenterVertically
            VStack(alignment: .leading, spacing: 0) { // 对应 Kotlin Column
                Text(marker.name)
                    .font(AppTokens.TypographyTokens.title)
                    .foregroundColor(AppTokens.Colors.onSurface)
                Text(marker.distance)
                    .font(AppTokens.TypographyTokens.body)
                    .foregroundColor(AppTokens.Colors.secondary)
            }
            Spacer() // 将按钮推到右侧
            Button(action: {
                // 处理 View 按钮操作
                print("View \(marker.name) Tapped")
            }) {
                Text("View")
                    .font(AppTokens.TypographyTokens.label)
                    .foregroundColor(AppTokens.Colors.onPrimary)
                    // 按钮内部填充，模拟 Compose Button 的默认填充
                    .padding(.horizontal, AppTokens.Spacing.md)
                    .padding(.vertical, AppTokens.Spacing.xs)
                    .background(AppTokens.Colors.primary)
                    .cornerRadius(AppTokens.Shapes.small)
            }
        }
        .padding(AppTokens.Spacing.md) // 对应 Kotlin Card 内部的 .padding(AppTokens.Spacing.md)
        .background(AppTokens.Colors.surface) // 对应 Kotlin Card 的 containerColor
        .cornerRadius(AppTokens.Shapes.medium) // 圆角
        .shadow(color: Color.black.opacity(AppTokens.ElevationMapping.level2.opacity), // 阴影颜色和透明度
                radius: AppTokens.ElevationMapping.level2.radius, // 阴影模糊半径
                y: AppTokens.ElevationMapping.level2.y) // 阴影 Y 轴偏移
        .frame(maxWidth: .infinity) // 填充宽度
    }
}

// MARK: - App Entry Point
// 对应 Kotlin 的 MainActivity 和 setContent
@main
struct SingleFileUIApp: App {
    var body: some Scene {
        WindowGroup {
            RootScreen()
        }
    }
}
